feat!: native `help` command used for opening

simonmandlik opened the pull request:
Here's my attempt at closing https://github.com/catgoose/telescope-helpgrep.nvim/issues/6

Unfortunately, the solution was not as straightforward as I described in https://github.com/catgoose/telescope-helpgrep.nvim/issues/6, because `help options.txt` works because `options.txt` is itself a tag and `help` doesn't work with any file.

Similarly to `Telescope help_tags` I build an index of helpfile-tag mappings, where for each help file I store one tag it contains. I can then open `help` using this tag and jump to the right location rightaway.

This PR will also open help in a new tab or a new vsplit (again inspired by `Telescope help_tags`). This has a nice advantage of reusing user settings (or defaults) for livegrep, and not having to define any mappings at all. On the other hand, this makes this a breaking change.

What do you think @catgoose?

simonmandlik committed:
feat!: native `help` command used for opening

catgoose commented:
What if there are no tags for that file?  For example if you search for `colorizer`:

![image](https://github.com/user-attachments/assets/b6b6a3b3-c954-41f8-add6-a23832d30572)

This does nothing when selecting the entry.  Could we fall back to editing `colorizer.txt` in this case?

I do sort of like being able to set keybindings for different pickers in `telescope.setup`:

![image](https://github.com/user-attachments/assets/b3c09070-49a4-409d-bcda-c66224f0d572)

This way I can have Telescope defaults, with overrides like above:

![image](https://github.com/user-attachments/assets/3c9cc2a5-3bae-45dc-8dbe-1e65e9aa8ad2)

simonmandlik committed:
fix!: fallback to the standard `edit` if the help file contains no tags

simonmandlik commented:
@catgoose please, take a look now

> For example if you search for colorizer:

Weird, when I'm browsing `nvim-colorizer.lua` docs, I can see the tags there and use them. But I added a fallback to the standard "grep" action.

> I do sort of like being able to set keybindings for different pickers

Added `mappings` back to config. I just changed to default to `{}`, i.e. keep all defaults from telescope. IMHO previously it made sense to not use `actions.select_default` for `<CR>` as it would replace the current buffer, however, `:help`, which is now called from `actions.select_default` won't do that.

catgoose commented:
Can you remove the `vim.print`?

simonmandlik committed:
feat!: added back `mappings` to config (changed the default to `{}`)

simonmandlik committed:
docs: added a note about extending opening actions to README.md

simonmandlik commented:
@catgoose oops, sorry, removed

catgoose commented:
@simonmandlik thanks, I like this!
